### ISSUE TITLE
jshint randomly shows prejudice against single quotes

### ISSUE THREAD

**Issue by AgentME** [jshint/jshint#2194]
The following code:

```javascript
['el', 'frameElement', 'menubar', 'opener', 'pr', 'sc', 'si', 'pos'].forEach(function(name) {
  Object.defineProperty(window, name, {
    set: function(x) {
      console.log('tried to set window.'+name+' to', x);
    }
  });
});
```

gets this output from jshint:

```
test.js: line 1, col 2, Strings must use doublequote.
test.js: line 1, col 8, Strings must use doublequote.
test.js: line 1, col 24, Strings must use doublequote.
test.js: line 1, col 35, Strings must use doublequote.
test.js: line 1, col 45, Strings must use doublequote.
test.js: line 1, col 51, Strings must use doublequote.
test.js: line 1, col 57, Strings must use doublequote.
test.js: line 1, col 63, Strings must use doublequote.

8 errors
```

If you remove an item from the list on line 1, you get no errors instead.

jshint v2.6.0

**Comment by mbildner** [comment]
Starting to look into this bug - so far it looks like it only occurs when the array is the first identifier on the page and `array.length` >= 8

which is super weird.

**Comment by rwaldron** [comment]
What version are you using? JSHint has deprecated style-centric rules, in favor of encouraging the use of JSCS. 

cc @mikesherov

**Comment by AgentME** [comment]
I'm using jshint 2.6.0 installed from NPM.

**Comment by mikesherov** [comment]
@rwaldron, perhaps it's time we discuss removal strategy?

**Comment by mbildner** [comment]
@rwaldron @mikesherov no I don't believe this is actually a style-checker issue, I think it's a real bug.

Addressed by this pr: https://github.com/jshint/jshint/pull/2200

The problem is the parser is being thrown into `JSON` mode when it shouldn't be, but it is right that in that mode it should throw errors at single quotes. I would love some insight into where the magic number 15 comes from:

https://github.com/jshint/jshint/blob/8a231e29422deddd8aef5e9bd65632a9bebf4528/src/jshint.js#L4612

I can't really imagine a situation where this would be needed -- or at least we can up the limit (>100).

**Comment by mbildner** [comment]
ironically even adding `'use strict'` would have kept this from being noticed :)

**Comment by lukeapage** [comment]
Released in 2.6.1
